src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs

```diff
@@ -191,17 +191,45 @@ public static class StandardLearnersCatalog
         }
 
         /// <summary>
-        /// Predict a target using a linear binary classification model trained with the AveragedPerceptron trainer.
+        /// Predict a target using a linear binary classification model trained with averaged perceptron trainer.
         /// </summary>
+        /// <remarks>
+        /// Perceptron is a classification algorithm that makes its predictions based on a linear function.
+        /// For instance with feature values f0, f1,..., f_D-1, the prediction is given by the sign of sigma[0, D-1] (w_i * f_i), where w_0, w_1,..., w_D-1 are the weights computed by the algorithm.
+        ///
+        /// Perceptron is an online algorithm, i.e., it processes the instances in the training set one at a time.
+        /// The weights are initialized to be 0, or some random values. Then, for each example in the training set, the value of sigma[0, D-1] (w_i * f_i) is computed.
+        /// If this value has the same sign as the label of the current example, the weights remain the same.If they have opposite signs,
+        /// the weights vector is updated by either subtracting or adding (if the label is negative or positive, respectively) the feature vector of the current example,
+        /// multiplied by a factor 0 &lt; a &lt;= 1, called the learning rate.In a generalization of this algorithm, the weights are updated by adding the feature vector multiplied by the learning rate,
+        /// and by the gradient of some loss function (in the specific case described above, the loss is hinge-loss, whose gradient is 1 when it is non-zero).
+        ///
+        /// In Averaged Perceptron (AKA voted-perceptron), the weight vectors are stored,
+        /// together with a weight that counts the number of iterations it survived (this is equivalent to storing the weight vector after every iteration, regardless of whether it was updated or not).
+        /// The prediction is then calculated by taking the weighted average of all the sums sigma[0, D-1] (w_i * f_i) or the different weight vectors.
+        ///
+        /// For more information see <a href="https://en.wikipedia.org/wiki/Perceptron">Wikipedia entry for Perceptron</a>
```

Comment:
Maybe move this to the main "Perceptron" paragraph above the "average perceptron" paragraph?

Reply:
this section is the ending references in case the description above wasn't enough. I think it's better to keep them at the end, like footnotes.

---
In reply to: [255264223](https://github.com/dotnet/machinelearning/pull/2483#discussion_r255264223) [](ancestors = 255264223)